### PR TITLE
Feature 6.1.x E2E tests

### PR DESCRIPTION
**Pull request by andriizhegurov-okta**
Feature script:

```
Feature: 6.1: Multi-Factor Authentication with Password and Email

  Background:
	Given a SPA, WEB APP or MOBILE Policy that defines MFA with Password and Email as required
	And a User named "Mary" created in the admin interface with a Password only

  Scenario: 6.1.2: 2FA Login with Email
	Given Mary navigates to the Basic Login View
# Added missing step
	And she has inserted her username
#
	And she has inserted her password
# ???
#	And her password is correct 
	When she clicks Login
	Then she is presented with an option to select Email to verify
	When She selects Email from the list

# not a real button
	And She selects "Receive a Code"
	Then the screen changes to receive an input for a code
	When She inputs the correct code from the Email
#	And She selects "Verify"
	Then she is redirected to the Root View
    And an application session is created

  Scenario: 6.1.3: Mary enters a wrong verification code
	Given Mary navigates to the Basic Login View
	When she fills in her username
	And she fills in her correct password
	And she clicks Login
	Then She sees a list of factors
	When She has selected Email from the list of factors
	And She inputs the incorrect code from the email
	Then the sample shows an error message "Invalid code. Try again." on the Sample App
```

### CODE REVIEW

File: samples/samples-aspnet/embedded-auth-with-sdk/embedded-auth-with-sdk/Controllers/AccountController.cs [hunk 1]
```diff
@@ -40,6 +40,11 @@ public async Task<ActionResult> Login(string returnUrl)
                 IdpOptions = identityProvidersResponse.IdpOptions,  // You can keep IdpOptions unset (set to null) if you don't want or need social login buttons
             };
 
+            if (TempData.ContainsKey("TerminalStateMessage"))
```
Comment (laura-rodriguez) [inline samples/samples-aspnet/embedded-auth-with-sdk/embedded-auth-with-sdk/Controllers/AccountController.cs:43]: I thought this change was already included in master 🤔

File: src/Okta.Idx.Sdk.E2ETests/Helpers/A18NClient/A18nClient.cs [hunk 2]
```diff
@@ -305,17 +305,17 @@ private string GetRelativeUri(string resource = default)
...
-                DeleteProfileAsync(profile.ProfileId).Wait();
-            }
-        }
+        }
+
+        private void CleanUpOldUsedProfiles(string newProfileUniqueTag)
+        {
+            var activeProfiles = GetActiveProfilesAsync().Result;
+
+            foreach (var profile in activeProfiles.Profiles.Where(p => p.DisplayName == newProfileUniqueTag))
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.E2ETests/Helpers/A18NClient/A18nClient.cs:314]: do we need a null check for profiles (`activeProfiles?.Profiles?`)?

File: src/Okta.Idx.Sdk.E2ETests/Steps/BaseTestSteps.cs [hunk 3]
```diff
@@ -4,12 +4,14 @@ namespace Okta.Idx.Sdk.E2ETests.Steps
 {
     public abstract class BaseTestSteps
     {
-        protected TestUserProperties _testUser;
+      //  protected TestUserProperties _testUser;
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.E2ETests/Steps/BaseTestSteps.cs:7]: Nit: Remove this line.

File: src/Okta.Idx.Sdk.E2ETests/Steps/MfaWithPasswordAndEmailSteps.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        [Given(@"a User named ""(.*)"" created in the admin interface with a Password only")]
+        public async Task GivenAUserNamedCreatedInTheAdminInterfaceWithAPasswordOnly(string firstName)
+        {
+            _testConfig.TestUser = await _userHelper.GetActivePasswordAndEmailUserAsync(firstName);
+        }
+
+        [Given(@"her password is correct")]
+        public void GivenHerPasswordIsCorrect()
+        {
+            //???
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.E2ETests/Steps/MfaWithPasswordAndEmailSteps.cs:29]: Do we need this method?

File: src/Okta.Idx.Sdk.E2ETests/Steps/Pages/HomePageSteps.cs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        public void GivenMaryNavigatesToTheLoginPage()
+        {
+            _homePageModel.GoToPage();
+            _homePageModel.AssertPageOpenedAndValid();
+            _homePageModel.LoginButton.Click();
+            _loginPageModel.AssertPageOpenedAndValid();
+        }
+
+        [Then(@"she is redirected to the Root View")]
+        [Then(@"she is redirected to the Root Page is provided")]
```
Comment (laura-rodriguez) [inline src/Okta.Idx.Sdk.E2ETests/Steps/Pages/HomePageSteps.cs:38]: `is provided` doesn't seem to make sense in this sentence 🤔

**Review by laura-rodriguez — APPROVED** [review]
Minor comments, other than that LGTM.

File: src/Okta.Idx.Sdk.E2ETests/Helpers/A18NClient/A18nClient.cs (hunk 2 above)
Comment (andriizhegurov-okta) [reply src/Okta.Idx.Sdk.E2ETests/Helpers/A18NClient/A18nClient.cs:314]: I don't see how they can be null, I expect this to be returned when there are no profiles available:
```json
{
    "profiles": [
    ]
}
```

File: src/Okta.Idx.Sdk.E2ETests/Steps/Pages/HomePageSteps.cs (hunk 5 above)
Comment (andriizhegurov-okta) [reply src/Okta.Idx.Sdk.E2ETests/Steps/Pages/HomePageSteps.cs:38]: I agree. It's from Feature 3.1, line 19. Another place to make edits to the cucumber scenario

File: src/Okta.Idx.Sdk.E2ETests/Steps/MfaWithPasswordAndEmailSteps.cs (hunk 4 above)
Comment (andriizhegurov-okta) [reply src/Okta.Idx.Sdk.E2ETests/Steps/MfaWithPasswordAndEmailSteps.cs:29]: I was thinking about the same, why would we need this test step: "her password is correct"?
I think I will remove this comment, but leave the function for now.

File: samples/samples-aspnet/embedded-auth-with-sdk/embedded-auth-with-sdk/Controllers/AccountController.cs (hunk 1 above)
Comment (andriizhegurov-okta) [reply samples/samples-aspnet/embedded-auth-with-sdk/embedded-auth-with-sdk/Controllers/AccountController.cs:43]: oh, it's in Master actually, need to update the branch

**Pull request merged**